File: labs/w9/sort.c
#include <stdio.h>

int bubbleSort(int *array, int size) {
   if ((array == NULL) || (size == 0)) {
      return -1;
   } else {
      int swapped = 1, j = 1;
      while (swapped) {
         swapped = 0;
         for (j=1;j<size;j++) {
            if (array[j-1] > array[j]) {
               int temp = array[j];
               array[j] = array[j-1];
               array[j-1] = temp;
               swapped = 1;
            }
         }
      }
      return 0;
   }
}

File: labs/w9/sort.py
def bubbleSort(array):
   if type(array) != list:
      return [False, array]

   for x in array:
      if type(x) != int:
         return [False, array]

   arr = list(array)
   n = len(arr)

   if n == 0:
      return [False, []]

   swapped = True

   while swapped:
      swapped = False
      for j in arr[1:n]:
         if arr[j-1] > arr[j]:
            temp = arr[j]
            arr[j] = arr[j-1]
            arr[j-1] = temp
            swapped = True

   return [True, arr]
